File: 2020/07/15-GoodPair.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer, Integer> counter = new HashMap<Integer, Integer>();
        
        for (int i = 0; i < nums.length; ++i) {
            if (counter.containsKey(nums[i])) {
                int temp = counter.get(nums[i]);
                counter.put(nums[i], (temp + 1));
            } else {
                counter.put(nums[i], 1);
            }
        }
        
        int res = 0;
        
        for (int j : counter.keySet()) {
            int temp = counter.get(j);
            res += ((temp * (temp - 1)) / 2);
        }
        
        return res;
    }
}

File: 2020/07/15-GoodPair.kt
class Solution {
    fun numIdenticalPairs(nums: IntArray): Int {
        var counter : HashMap<Int, Int> = HashMap<Int, Int>()
        
        for (num in nums) {
            if (counter.containsKey(num)) {
                var temp : Int = counter[num]!!
                counter[num] = ++temp
            } else {
                counter.put(num, 1)
            }
        }
        
        var res : Int = 0
        
        for (key in counter.keys) {
            var temp : Int = counter[key]!!
            res += ((temp * (temp - 1)) / 2)
        }
        return res
    }
}